packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -222,20 +222,74 @@ export function includeDerivedModel(model: Model): boolean {
   );
 }
 
+function applyEncoding(
+  dpgContext: SdkContext,
+  typespecType: Scalar | ModelProperty,
+  target: any = {}
+) {
+  const encodeData = getEncode(dpgContext.program, typespecType);
+  if (encodeData) {
+    const newTarget = { ...target };
+    const newType = getSchemaForScalar(dpgContext, encodeData.type);
+    newTarget["type"] = newType["type"];
+    newTarget["typeName"] = newType["typeName"];
+    newTarget["outputTypeName"] = newType["outputTypeName"];
+    // If the target already has a format it takes priority. (e.g. int32)
+    newTarget["format"] = mergeFormatAndEncoding(
+      newTarget.format,
+      encodeData.encoding,
+      newType["format"]
+    );
+    return newTarget;
+  }
+  return target;
+}
+
+function mergeFormatAndEncoding(
+  format: string | undefined,
+  encoding: string,
+  encodeAsFormat: string | undefined
+): string {
+  switch (format) {
+    case undefined:
+      return encodeAsFormat ?? encoding;
+    case "date-time":
+      return encoding;
+    case "duration":
+    default:
+      return encodeAsFormat ?? encoding;
+  }
+}
+
 function getSchemaForScalar(
   dpgContext: SdkContext,
   scalar: Scalar,
   relevantProperty?: ModelProperty
 ) {
-  const program = dpgContext.program;
-  const encodeData = getEncode(program, scalar);
-  let result: any = encodeData
-    ? getSchemaForScalar(dpgContext, encodeData.type)
-    : getSchemaForStdScalar(program, scalar, relevantProperty);
-  if (!result && scalar.baseScalar) {
+  let result = {};
+  const isStd = dpgContext.program.checker.isStdType(scalar);
+  if (isStd) {
+    result = getSchemaForStdScalar(
+      dpgContext.program,
+      scalar,
+      relevantProperty
+    );
+  } else if (scalar.baseScalar) {
     result = getSchemaForScalar(dpgContext, scalar.baseScalar);
   }
-  return applyIntrinsicDecorators(program, scalar, result);
+  const withDecorators = applyEncoding(
+    dpgContext,
+    scalar,
+    result
+      ? applyIntrinsicDecorators(dpgContext.program, scalar, result)
+      : undefined
+  );
+  if (withDecorators.type === "string" && withDecorators.format === "binary") {
```

Comment:
Can this be the generic handling for format equals binary?  Any other types that we need special handling?

Reply:
First of all, bytes with any other encodings like base64 or base64url are just pure string in RLC, we only need to special handle binary bytes in both standard scalar and non-standard scalar. this is the place where we handle non-standard scalar and I also add similar logic in standard scalar here https://github.com/Azure/autorest.typescript/pull/2056/files#diff-b9d6429c3d7eab57015f06928f591a73117fa57b586c4eb7f1bdd7c1ec2e16d9R1047-R1057 
Secondly, the type we get from typespec for `@encode("binary")` is string no matter what the original type is. 
if we have something like 
```
  @encode("binary")
  weight: int32;
```
we also get in OpenAPI3 emitter.
```yaml
        weight:
          type: string
          format: binary
```